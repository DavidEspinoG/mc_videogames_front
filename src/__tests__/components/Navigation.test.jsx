import userEvent from '@testing-library/user-event';
import axios from 'axios';
import Navigation from '../../components/navigation';
import renderWithProviders from '../test-utils';

describe('Navigation', () => {
  it("should render as expected when there's no logged in user", () => {
    const { container } = renderWithProviders(<Navigation />);
    expect(container).toMatchSnapshot();
  });

  it('should render as expected when there is a logged in user', () => {
    const { container } = renderWithProviders(<Navigation />, {
      preloadedState: {
        user: {
          user: {
            name: 'user name',
          },
        },
      },
    });
    expect(container).toMatchSnapshot();
  });

  it('should render as expected when there is a logged in admin', () => {
    const { container } = renderWithProviders(<Navigation />, {
      preloadedState: {
        user: {
          user: {
            name: 'admin name',
            admin: true,
          },
        },
      },
    });
    expect(container).toMatchSnapshot();
  });

  it('should toggle css classes when clicking the toggler button', async () => {
    const { getByTestId } = renderWithProviders(<Navigation />);
    expect(getByTestId('navbar-collapse')).toHaveClass('desktop-show');
    await userEvent.click(getByTestId('navbar-toggler-button'));
    expect(getByTestId('navbar-collapse')).toHaveClass('show');
    await userEvent.click(getByTestId('navbar-toggler-button'));
    expect(getByTestId('navbar-collapse')).toHaveClass('desktop-show');
  });

  it('should collapse the navbar when clicking any navLink (e.g. Home)', async () => {
    const { getByRole, getByTestId } = renderWithProviders(<Navigation />);
    expect(getByTestId('navbar-collapse')).toHaveClass('desktop-show');
    await userEvent.click(getByTestId('navbar-toggler-button'));
    expect(getByTestId('navbar-collapse')).toHaveClass('show');
    await userEvent.click(getByRole('link', { name: 'Home' }));
    expect(getByTestId('navbar-collapse')).toHaveClass('desktop-show');
  });

  it('should remove user from the store when clicking the logout button', async () => {
    const { store, getByRole } = renderWithProviders(<Navigation />, {
      preloadedState: {
        user: {
          user: {
            name: 'user name',
          },
        },
      },
    });

    axios.delete.mockResolvedValue({ status: 200, data: {} });
    expect(store.getState().user.user).toBeTruthy();
    await userEvent.click(getByRole('button', { name: 'Log out' }));
    expect(store.getState().user.user).toBeFalsy();
  });
});
