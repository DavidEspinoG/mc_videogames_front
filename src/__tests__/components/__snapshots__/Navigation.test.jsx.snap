// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation > should render as expected when there is a logged in admin 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="nav-container"
    >
      <div
        class="nav-group"
      >
        <div
          class="nav-header flex-md-column"
        >
          <button
            class="navbar-toggler-button"
            data-testid="navbar-toggler-button"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="user-name order-md-1"
          >
            <h3
              class="welcome mb-0 fw-bolder"
            >
              Welcome,
               
              admin name
            </h3>
          </div>
          <a
            class="nav-logo d-flex justify-content-center"
            href="/"
          >
            <img
              alt="logo"
              class="logo"
              src="/src/assets/logo.webp"
            />
          </a>
        </div>
        <div
          class="collapse navbar-collapse desktop-show"
          data-testid="navbar-collapse"
        >
          <ul
            class="navbar-nav"
          >
            <li>
              <a
                aria-current="page"
                class="nav-link active"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class="nav-link"
                href="/reserve"
              >
                Reserve
              </a>
            </li>
            <li>
              <a
                class="nav-link"
                href="/myReservations"
              >
                My reservations
              </a>
            </li>
            <li>
              <a
                class="nav-link"
                href="/add"
              >
                Add
              </a>
            </li>
            <li>
              <a
                class="nav-link"
                href="/delete"
              >
                Delete
              </a>
            </li>
            <li>
              <button
                class="nav-link logout-button"
                type="button"
              >
                Log out
              </button>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="credits"
      >
        <small>
           © 2023 - All rights reserved
        </small>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Navigation > should render as expected when there is a logged in user 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="nav-container"
    >
      <div
        class="nav-group"
      >
        <div
          class="nav-header flex-md-column"
        >
          <button
            class="navbar-toggler-button"
            data-testid="navbar-toggler-button"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="user-name order-md-1"
          >
            <h3
              class="welcome mb-0 fw-bolder"
            >
              Welcome,
               
              user name
            </h3>
          </div>
          <a
            class="nav-logo d-flex justify-content-center"
            href="/"
          >
            <img
              alt="logo"
              class="logo"
              src="/src/assets/logo.webp"
            />
          </a>
        </div>
        <div
          class="collapse navbar-collapse desktop-show"
          data-testid="navbar-collapse"
        >
          <ul
            class="navbar-nav"
          >
            <li>
              <a
                aria-current="page"
                class="nav-link active"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class="nav-link"
                href="/reserve"
              >
                Reserve
              </a>
            </li>
            <li>
              <a
                class="nav-link"
                href="/myReservations"
              >
                My reservations
              </a>
            </li>
            <li>
              <button
                class="nav-link logout-button"
                type="button"
              >
                Log out
              </button>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="credits"
      >
        <small>
           © 2023 - All rights reserved
        </small>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Navigation > should render as expected when there's no logged in user 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="nav-container"
    >
      <div
        class="nav-group"
      >
        <div
          class="nav-header flex-md-column"
        >
          <button
            class="navbar-toggler-button"
            data-testid="navbar-toggler-button"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <a
            class="nav-logo d-flex justify-content-center"
            href="/"
          >
            <img
              alt="logo"
              class="logo"
              src="/src/assets/logo.webp"
            />
          </a>
        </div>
        <div
          class="collapse navbar-collapse desktop-show"
          data-testid="navbar-collapse"
        >
          <ul
            class="navbar-nav"
          >
            <li>
              <a
                aria-current="page"
                class="nav-link active"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class="nav-link"
                href="/login"
              >
                Login
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="credits"
      >
        <small>
           © 2023 - All rights reserved
        </small>
      </div>
    </div>
  </nav>
</div>
`;
